Диплом 1

Olhha opened the pull request:
- добавлен Сервер Статистики + клиент
- Main service - загрушки для двух эндпоинтов /events, /events/{id}, в которых реализована запись статистики

Olhha committed:
Диплом 1:
- добавлен Сервер Статистики + клиент
- Main service - загрушки для двух эндпоинтов /events, /events/{id}, в которых реализована запись статистики

AnastasiyaFomkina commented on main-service/src/main/java/ru/practicum/controller/MainController.java:
```diff
@@ -0,0 +1,46 @@
+
+@RestController
+@RequestMapping("/events")
+public class MainController {
```
Забегу немного вперед и скажу, что всю работу с модулем статистики будет удобнее выделить в отдельный сервис статистики в модуле main. Данный контроллер на следующих этапах никак не понадобится, поэтому его лучше удалить

Olhha replied:
да, конечно, я добавила его только для тестирования статистики

AnastasiyaFomkina commented on stats-service/stats-server/src/main/java/ru/practicum/exception/ErrorHandler.java:
```diff
@@ -0,0 +1,55 @@
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    public ErrorResponse handleDateTimeParseException(final DateTimeParseException e) {
+        log.debug("Validation Exception, статус 400 Bad request: {}", e.getMessage(), e);
+        return new ErrorResponse("Формат даты должен быть: yyyy-MM-dd HH:mm:ss ." + e.getMessage());
```
Паттерн нужно вынести в публичную константу и переиспользовать. Чтобы быстро поправить все значения, можно воспользоваться горячими клавишами в IDEA: [Ctrl+Shift+R](https://www.jetbrains.com/help/idea/finding-and-replacing-text-in-project.html?keymap=primary_windows#replace_search_string_in_project)

Olhha replied:
done

AnastasiyaFomkina commented on stats-service/stats-server/src/main/java/ru/practicum/model/EndpointHit.java:
```diff
@@ -0,0 +1,28 @@
+public class EndpointHit {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    Long id;
```
Полям нужно добавить модификаторы доступа

Olhha replied:
done

Olhha committed:
Диплом 1:
- удалила MainController
- вынесла формат Даты в утилитарный класс DateTimePattern
- мелкие исправления
